v1/src/test/java/com/google/cloud/teleport/templates/TextToPubsubStreamLT.java

```diff
@@ -0,0 +1,220 @@
+  }
+
+  @After
+  public void teardown() {
+    ResourceManagerUtils.cleanResources(pubsubResourceManager, gcsClient);
+  }
+
+  @Test
+  public void test1Backlog10gb() throws IOException, ParseException, InterruptedException {
+    testBacklog(Function.identity(), NUM_MESSAGES_FOR_10GB, TIMEOUT_FOR_10_GB_TEST_MINUTES);
+  }
+
+  @Test
+  public void test2SteadyState1hr() throws IOException, ParseException, InterruptedException {
+    testSteadyState1hr(Function.identity(), TIMEOUT_FOR_1_HOUR_TEST_MINUTES);
+  }
+
+  @Test
+  public void test3Backlog100gb() throws IOException, ParseException, InterruptedException {
+    testBacklog(Function.identity(), NUM_MESSAGES_FOR_100GB, TIMEOUT_FOR_100_GB_TEST_MINUTES);
+  }
+
+  private void testBacklog(
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder,
+      long numMessages,
+      long timeout)
+      throws IOException, ParseException, InterruptedException {
+    // Creates a topic on Pub/Sub.
+    TopicName outputTopic = pubsubResourceManager.createTopic("topic");
+    // Creates a subscription on Pub/Sub.
+    SubscriptionName outputSubscription =
+        pubsubResourceManager.createSubscription(outputTopic, "output-subscription");
+
+    DataGenerator dataGenerator =
+        DataGenerator.builderWithSchemaTemplate(testName, "GAME_EVENT")
+            .setQPS("1000000")
+            .setMessagesLimit(String.valueOf(numMessages))
+            .setSinkType("GCS")
+            .setOutputDirectory(getTestMethodDirPath())
+            .setNumShards("20")
+            .setNumWorkers("50")
+            .setMaxNumWorkers("100")
+            .build();
+    // Executes the data generator
+    dataGenerator.execute(Duration.ofMinutes(timeout));
+
+    // Wait for the template to start
+    TimeUnit.SECONDS.sleep(WAIT_FOR_GENERATOR_TIME_SEC);
+
+    LaunchConfig options =
+        paramsAdder
+            .apply(
+                LaunchConfig.builder(testName, SPEC_PATH)
+                    .addEnvironment("maxWorkers", 100)
+                    .addParameter("outputTopic", outputTopic.toString())
+                    .addParameter("inputFilePattern", getTestMethodDirPath() + "/*"))
+            .build();
+    // Act
+    PipelineLauncher.LaunchInfo info = pipelineLauncher.launch(project, region, options);
+    assertThatPipeline(info).isRunning();
+
+    // Waits until a number of messages condition is met
+    PipelineOperator.Result result =
+        pipelineOperator.waitForConditionAndFinish(
+            createConfig(info, Duration.ofMinutes(timeout)),
+            () -> waitForNumMessages(info.jobId(), INPUT_PCOLLECTION, numMessages));
+
+    // Assert
+    assertThatResult(result).meetsConditions();
+
+    // export results
+    exportMetricsToBigQuery(info, getMetrics(info, INPUT_PCOLLECTION, OUTPUT_PCOLLECTION));
+  }
+
+  private void testSteadyState1hr(
+      Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder, long timeout)
```

Comment:
do we need to have a timeout argument here? 1 hour is always 60 minutes, it's a constant. Let's set the timeout in `Duration.ofMinutes(TIMEOUT_FOR_1_HOUR_TEST_MINUTES)` instead of `Duration.ofMinutes(timeout)`. Just to remove unnecessary arguments.